src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/DependencyInjectionEventSourceTests.cs

```diff
@@ -17,6 +17,28 @@ public class EventSourceTests : ICollectionFixture<EventSourceTests>
     {
     }
 
+    internal class TestEventListener : EventListener
```

Comment:
I'm not a fan of this because the events are global. Any other tests that uses the event source can clash with yours.